tests/bls_onchain_test.go

```diff
@@ -0,0 +1,248 @@
+package tests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/0xsoniclabs/sonic/scc/bls"
+	"github.com/0xsoniclabs/sonic/tests/contracts/blsContracts"
+	gnark "github.com/consensys/gnark-crypto/ecc/bls12-381"
+	"github.com/consensys/gnark-crypto/ecc/bls12-381/fp"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBlsVerificationOnChain(t *testing.T) {
+	net := StartIntegrationTestNet(t, IntegrationTestNetOptions{
+		FeatureSet: opera.AllegroFeatures,
+	})
+	defer net.Stop()
+
+	// Deploy contract with transaction options
+	txOptions, err := net.GetTransactOptions(&net.account)
+	require.NoError(t, err, "failed to get transact options; %v", err)
+	txOptions.Nonce = nil
+	txOptions.GasLimit = 10000000
+	blsContract, _, err := DeployContractWithOpts(net, blsContracts.DeployBLS, txOptions)
+	require.NoError(t, err, "failed to deploy contract; %v", err)
+
+	// Test different bls libraries verification
+	tests := []struct {
+		name     string
+		detaFunc func(pk bls.PrivateKey, message []byte) ([]byte, []byte, error)
+	}{
+		{"gnark", getGnarkData},
+		{"bls", getBlsData},
```

Comment:
If I understand correctly, both of these data sources are required to produce exactly the same data for a given private key and message. Checking this, is beyond the scope of an integration test. This would be better placed in an unit test for the BLS library.

I think in this context you can pick one of these, simplifying the structure of this test.